### PR TITLE
Updates to use GRIB2 template 4.9 for QPFFFG and QPFARI variables in RRFS

### LINKED ISSUE #905: Update GRIB2 encoding for QPFARI and QPFFFG variables
For the new QPFARI and QPFFFG fields in RRFS, we need the capability to encode them as probability of exceeding a threshold, which requires use of GRIB2 product definition template 4.9:

https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-1-1.shtml
https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_temp4-9.shtml

### LINKED ISSUE #955: Add instantaneous clear sky downward solar irradiance to RRFS
This field is produced in NAM-Nest, and required for RRFS.  There are PRs being merged to get the new netCDF field output in the RRFS history files.

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
In order to appropriately encode the new QPFFFG and QPFARI variables as probabilities of QPF exceeding certain thresholds, we need to use GRIB2 template 4.9.  

This change requires an update to NCEPLIBS-g2tmpl...and a PR for the required changes has been opened:

https://github.com/NOAA-EMC/NCEPLIBS-g2tmpl/pull/117

The code was tested for RRFS_NA_3km on Jet.  

Combine another update:
output of the instantaneous clear-sky downward shortwave flux at the surface for RRFS, which is required since this field was output in NAM-Nest.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA Can you sync your branch with the latest UPP develop branch?

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAAA All flat control files besides RRFS need to be recreated since the updates in POST-XML-Library-NT.pl.
you might do:
```
cd UPP/parm
make clean
make
```

**Comment by WenMeng-NOAA** [comment]
@EricJames-NOAA I conducted the UPP standalone test with model files provided by you at /scratch1/BMC/wrfruc/ejames/wen_clearsky/
My test results are at /home/Wen.Meng/stmp2/rrfs_2023072903 on Hera.
I saw instant clear-sky SW downward flux generated as:
```
822:617809883:ndata=1905141:undef=0:mean=0.223535:min=0:max=48.45:cos_wt_mean=0.193224:d=2023072903:CSDSF:surface:1 hour fcst:
```
Please validate my test. Thanks!

**Comment by WenMeng-NOAA** [comment]
The UPP RTs have been completed on WCOSS2 and Hera. The baseline updates are needed for fv3r and 3drtma.
```
Wen.Meng@hfe07 stmp2$ pwd
/home/Wen.Meng/stmp2
Wen.Meng@hfe07 stmp2$ ls -ltr */*diff
-rw-r--r-- 1 Wen.Meng stmp   75 May 10 16:17 rap_2020072316_pe_test/WRFPRS.GrbF16.diff
-rw-r--r-- 1 Wen.Meng stmp  616 May 10 16:18 fv3r_2023062800/PRSLEV10.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp  616 May 10 16:19 fv3r_2023062800_pe_test/PRSLEV10.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 1086 May 10 16:19 rtma_2023040400_pe_test/PRSLEV00.tm00.diff
-rw-r--r-- 1 Wen.Meng stmp 1086 May 10 16:19 rtma_2023040400/PRSLEV00.tm00.diff
```

**Comment by EricJames-NOAA** [comment]
> @EricJames-NOAA I conducted the UPP standalone test with model files provided by you at /scratch1/BMC/wrfruc/ejames/wen_clearsky/ My test results are at /home/Wen.Meng/stmp2/rrfs_2023072903 on Hera. I saw instant clear-sky SW downward flux generated as:
> 
> ```
> 822:617809883:ndata=1905141:undef=0:mean=0.223535:min=0:max=48.45:cos_wt_mean=0.193224:d=2023072903:CSDSF:surface:1 hour fcst:
> ```
> 
> Please validate my test. Thanks!

@WenMeng-NOAA your results are identical to mine.  Thanks!

**Comment by WenMeng-NOAA** [comment]
> > @EricJames-NOAA I conducted the UPP standalone test with model files provided by you at /scratch1/BMC/wrfruc/ejames/wen_clearsky/ My test results are at /home/Wen.Meng/stmp2/rrfs_2023072903 on Hera. I saw instant clear-sky SW downward flux generated as:
> > ```
> > 822:617809883:ndata=1905141:undef=0:mean=0.223535:min=0:max=48.45:cos_wt_mean=0.193224:d=2023072903:CSDSF:surface:1 hour fcst:
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please validate my test. Thanks!
> 
> @WenMeng-NOAA your results are identical to mine. Thanks!

@EricJames-NOAA Thanks for confirming. I will warp up this PR for the merging process.

File: modulefiles/jet.lua [hunk 1]
```diff
@@ -14,6 +14,7 @@ cmake_ver=os.getenv("cmake_ver") or "3.23.1"
 load(pathJoin("cmake", cmake_ver))
 
 load("upp_common")
+load("g2tmpl")
```
Comment (WenMeng-NOAA) [inline modulefiles/jet.lua]: @EricJames-NOAA One more comment. Could you remove this load on jet?

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**